### PR TITLE
create the User class, controller, repository, and service

### PR DESCRIPTION
**Pull request by Zhangjason308**

### CODE REVIEW

**Comment by Nervith** [comment]
your UserTest.java crashes the compilation for spring boot, removing it allows it to compile. please resolve thanks.

**Comment by Zhangjason308** [comment]
> your UserTest.java crashes the compilation for spring boot, removing it allows it to compile. please resolve thanks.

You're best bet is to just comment out the tests for now as I have a PR up right now thats still in progress, so the changes wont be in until this PR goes in.

I'll remove the test cases though so everybody can work on their own tests

File: client/src/components/Home.svelte [hunk 1]
```diff
@@ -1,13 +1,17 @@
 <script>
   import { onMount } from "svelte";
 
+
+
```
Comment (nicholasyung1) [inline client/src/components/Home.svelte:5]: Remove extra spacing

File: server/src/main/java/Main.java [hunk 2]
```diff
@@ -109,7 +109,7 @@ private static void uploadPerk()
         }
 
         Perk newPerk = new Perk(perkValue, new ArrayList<>(), perkCode, product, membership);
-        perkManager.addPerk(newPerk);
+        //perkManager.addPerk(newPerk);
```
Comment (Triguy12) [inline server/src/main/java/Main.java:112]: Overall CLI from main file should be removed.

File: client/src/components/Profile.svelte [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+    }
+
+    .btn-primary {
+        width: 48%; /* Make buttons consistent */
+        padding: 0.75rem 1.5rem;
+        background-color: coral; /* Warm orange */
+        color: whitesmoke;
+        border: none;
+        border-radius: 5px;
+        font-size: 16px;
```
Comment (nicholasyung1) [inline client/src/components/Profile.svelte:268]: should this be 14 px to match the rest? (we have 28 and 14, I think is the only 16)

**Review by Triguy12 — APPROVED** [review]
Overall everything looks good. I believe there might be routing issues due to flipping between /api/desiredLocation and just /desiredLocation. I am going to approve this PR so I can run tests in my dev-branch.

File: server/src/main/java/Main.java
Comment (nicholasyung1) [inline server/src/main/java/Main.java:1]: thank you for these changes!

**Pull request merged**